### ISSUE TITLE
Error during SonarQube Scanner execution on typescript analysis

### ISSUE THREAD

**Issue by olsh** [SonarSource/SonarTS#572]
**SonarTS version:** 1.6 or 1.5, probably earlier versions too
**Node.js version:** 8.9.4
**TypeScript version:** 2.2.2

**SonarQube version:** 6.7.2 LTS
**SonarLint version and used IDE:** TFS extension v 4
**TSLint version:** 4.5.1

**Logs**
```
06:44:44.010 INFO: ------------------------------------------------------------------------
06:44:44.010 INFO: EXECUTION FAILURE
06:44:44.010 INFO: ------------------------------------------------------------------------
06:44:44.010 INFO: Total time: 36.171s
06:44:44.135 INFO: Final Memory: 52M/445M
06:44:44.135 INFO: ------------------------------------------------------------------------
06:44:44.135 ERROR: Error during SonarQube Scanner execution
java.lang.IllegalArgumentException: Start pointer [line=190, lineOffset=9] should be before end pointer [line=190, lineOffset=9]
	at org.sonar.api.internal.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.newRangeValidPointers(DefaultInputFile.java:325)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.newRange(DefaultInputFile.java:263)
	at org.sonar.plugin.typescript.SensorContextUtils.saveIssue(SensorContextUtils.java:82)
	at org.sonar.plugin.typescript.SensorContextUtils.saveIssues(SensorContextUtils.java:48)
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.analyze(ExternalTypescriptSensor.java:141)
	at org.sonar.plugin.typescript.ExternalTypescriptSensor.execute(ExternalTypescriptSensor.java:109)
	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:88)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:82)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:68)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:88)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:180)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:288)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:283)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:281)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:261)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
06:44:44.135 ERROR: 
06:44:44.135 ERROR: Re-run SonarQube Scanner using the -X switch to enable full debug logging.
06:44:44.135 DEBUG: Execution getVersion
06:44:44.135 DEBUG: Execution stop
Process returned exit code 1
The SonarQube Scanner did not complete successfully
```

We run the analysis from TFS using the TFS extension.
This happens on typescript analysis, if I disable it, the analysis completes with no errors.

**Comment by saberduck** [comment]
@olsh thanks for reporting the issue. Is it possible for you to share the source file on which the analysis is failing?

**Comment by olsh** [comment]
Hi @saberduck,

Thank you for the quick response!
I don't know on which file the analysis fails. Please check the first part of the logs and let me know which file I should send:

```
/2018-02-28T11:44:14.1986443Z 06:44:14.198 INFO: Sensor SonarTS [typescript]
2018-02-28T11:44:14.2134517Z 06:44:14.213 DEBUG: TypeScript compiler is found in this directory F:\TFS\Agentv2\_work\2\s\src\MyProject.Web.Admin\node_modules
2018-02-28T11:44:14.2134517Z 06:44:14.213 DEBUG: It will be used for analysis of typescript files
2018-02-28T11:44:14.2134517Z 06:44:14.213 DEBUG: Deploying bundle from `/sonarts-bundle.zip` to `F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0`
2018-02-28T11:44:14.6040865Z 06:44:14.604 DEBUG: Checking node version
2018-02-28T11:44:14.6353282Z 06:44:14.635 DEBUG: Using Node.js v8.9.4
2018-02-28T11:44:14.6509542Z 06:44:14.650 DEBUG: Analyzing 225 typescript file(s) with the following configuration file F:\TFS\Agentv2\_work\2\s\src\MyProject.Web.Admin\tsconfig.json
2018-02-28T11:44:14.7290791Z 06:44:14.729 DEBUG: Setting 'NODE_PATH' to F:\TFS\Agentv2\_work\2\s\src\MyProject.Web.Admin\node_modules
2018-02-28T11:44:14.7290791Z 06:44:14.729 DEBUG: Starting external process `node --max-old-space-size=2048 F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\bin\tsrunner`
2018-02-28T11:44:36.0884507Z 06:44:36.072 ERROR: Rule error during analysis of : F:/TFS/Agentv2/_work/2/s/src/MyProject.Web.Admin/angularApp/app/helpers/dx-data-grid/date-filter.ts TypeError: node.forEachChild is not a function
2018-02-28T11:44:36.0884507Z 06:44:36.072 ERROR:     at visitNode (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\rules\noReturnTypeAnyRule.js:64:22)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.getAllReturns (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\rules\noReturnTypeAnyRule.js:67:9)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.visitFunctionDeclaration (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\rules\noReturnTypeAnyRule.js:50:34)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.visitFunctionLikeDeclaration (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\utils\visitor.js:295:22)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.visitNode (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\utils\visitor.js:560:22)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at ts.forEachChild.child (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\utils\visitor.js:567:45)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at visitEachNode (F:\TFS\Agentv2\_work\2\s\src\MyProject.Web.Admin\node_modules\typescript\lib\typescript.js:14209:30)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Object.forEachChild (F:\TFS\Agentv2\_work\2\s\src\MyProject.Web.Admin\node_modules\typescript\lib\typescript.js:14382:24)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.visitChildren (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\utils\visitor.js:567:12)
2018-02-28T11:44:36.0884507Z 06:44:36.088 ERROR:     at Visitor.visitSourceFile (F:\TFS\Agentv2\_work\2\.sonarqube\out\.sonar\my-project_my-project_99279475-BE89-4DC3-AD47-9A9D139915B0\sonarts-bundle\node_modules\tslint-sonarts\lib\utils\visitor.js:239:14)
2018-02-28T11:44:43.6197012Z 06:44:43.619 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/about/about.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.6666128Z 06:44:43.666 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/about/about.routes.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.6666128Z 06:44:43.666 DEBUG: Not enough content in 'angularApp/app/about/about.routes.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.6666128Z 06:44:43.666 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/about/components/about.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.6822015Z 06:44:43.666 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.6978300Z 06:44:43.697 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents.routes.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7134548Z 06:44:43.713 DEBUG: Not enough content in 'angularApp/app/agents/agents.routes.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.7134548Z 06:44:43.713 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/agents.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7134548Z 06:44:43.713 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/agents.routes.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7134548Z 06:44:43.713 DEBUG: Not enough content in 'angularApp/app/agents/agents/agents.routes.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.7134548Z 06:44:43.713 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/components/agents.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7290773Z 06:44:43.729 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/components/agents/agent.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7290773Z 06:44:43.729 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/components/profile.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7447009Z 06:44:43.744 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/components/profile/groups.tab.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7447009Z 06:44:43.744 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/agents/agents/components/profile/profile.tab.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7606969Z 06:44:43.760 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/app.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7606969Z 06:44:43.760 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/app.constants.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7606969Z 06:44:43.760 DEBUG: Not enough content in 'angularApp/app/app.constants.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.7606969Z 06:44:43.760 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/app.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7759522Z 06:44:43.775 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/app.routes.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7759522Z 06:44:43.775 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/breadcrumbs.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7759522Z 06:44:43.775 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/_shared/shared.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7759522Z 06:44:43.775 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/clients.module.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7759522Z 06:44:43.775 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/clients.routes.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7935496Z 06:44:43.775 DEBUG: Not enough content in 'angularApp/app/clients/clients.routes.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.7935496Z 06:44:43.793 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/components/address.display.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.7935496Z 06:44:43.793 DEBUG: Not enough content in 'angularApp/app/clients/components/address.display.component.ts' to have CPD blocks, it will not be part of the duplication detection
2018-02-28T11:44:43.7935496Z 06:44:43.793 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/components/clients.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8228325Z 06:44:43.822 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/components/forms/client.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8228325Z 06:44:43.822 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/components/scratchcards_gen.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8384578Z 06:44:43.838 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/billing.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8697061Z 06:44:43.869 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/billing/payment-profiles.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8697061Z 06:44:43.869 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/dealer-tab/accesscard-list.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8697061Z 06:44:43.869 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/dealer-tab/assign-cards.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.8853282Z 06:44:43.885 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/dealer-tab/billing.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9167236Z 06:44:43.916 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/dealer-tab/client-list.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9167236Z 06:44:43.916 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/dealer.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9167236Z 06:44:43.916 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/address.form-master.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9322021Z 06:44:43.932 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/cancel-service.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9322021Z 06:44:43.932 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/detach-device.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9322021Z 06:44:43.932 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/payment-profile.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9478263Z 06:44:43.947 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/payment.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9478263Z 06:44:43.947 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/refund.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9634530Z 06:44:43.963 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/select-device.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9790756Z 06:44:43.963 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/forms/subscription.form.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9790756Z 06:44:43.979 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/main.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9790756Z 06:44:43.979 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/profile.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9947026Z 06:44:43.994 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/service-checkout/models/checkoutFormData.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:43.9947026Z 06:44:43.994 DEBUG: 'src/MyProject.Web.Admin/angularApp/app/clients/profile/components/service-checkout/summary.component.ts' generated metadata  with charset 'UTF-8'
2018-02-28T11:44:44.0103275Z 06:44:44.010 INFO: ------------------------------------------------------------------------
2018-02-28T11:44:44.0103275Z 06:44:44.010 INFO: EXECUTION FAILURE
2018-02-28T11:44:44.0103275Z 06:44:44.010 INFO: ------------------------------------------------------------------------

~~~~~~~~~~~~
Logs from the first comment
https://github.com/SonarSource/SonarTS/issues/572#issue-300996034
```

**Comment by saberduck** [comment]
hm, that's strange, because such code doesn't even compile with TypeScript compiler, so the error would  be parse error. But it the logs error is different. Can you share whole `date-filter.ts` ?

**Comment by olsh** [comment]
> hm, that's strange, because such code doesn't even compile with TypeScript compiler, so the error would be parse error.

Just tried to compile, it compiles with the error:
`error TS1005: 'finally' expected.`

Typescript
```ts
try {
  console.log("test");
}
```

And produces the following file:
Javascript
```js
try {
  

console.log("test");
}
finally {
}
```

date-filter.js
```ts
import * as moment0 from 'moment';
import { DxDataGridComponent } from 'devextreme-angular';

const moment = moment0;

export function calculateFilterExpression(value: any, selectedFilterOptions: any, target: any): any {
    if (selectedFilterOptions === 'between') {
        const values: any[] = [];
        for (let val of value) {
            const date: any = moment(val);
            val = date.add(-date.utcOffset(), 'minutes').toDate();
            values.push(val);
        }

        arguments[0] = values;
        let r = this.defaultCalculateFilterExpression.apply(this, arguments);
        return r;
    } else if (selectedFilterOptions === '=' || selectedFilterOptions === '<>') {
        let r = this.defaultCalculateFilterExpression.apply(this, arguments);
        let date = moment(r[0][2]);
        r[0][2] = date.add(-date.utcOffset(), 'minutes').toDate();
        date = moment(r[2][2]);
        r[2][2] = date.add(-date.utcOffset(), 'minutes').toDate();
        return r;
    } else {
        let r = this.defaultCalculateFilterExpression.apply(this, arguments);
        let date = moment(value);
        r[2] = date.add(-date.utcOffset(), 'minutes').toDate();
        return r;
    }
}

export function fixUtcDateFilterDxDataGrid(dxDataGrid: DxDataGridComponent, exceptColumnNames: string[] = []): void {
    for (let column of dxDataGrid.columns) {
        if (column.dataType === 'date') {

            if (exceptColumnNames != null && exceptColumnNames.find(c => c === column.dataField)) {
                continue;
            }

            if (column.calculateFilterExpression != null) {
                continue;
            }

            column.calculateFilterExpression = calculateFilterExpression;
        }
    }
}
```

**Comment by vilchik-elena** [comment]
Hi @olsh, you are not supposed to analyse files with parsing errors.
In this case AST is generated as usual and rules are executed on it, but this AST might lack some parts and you'll have runtime problems. So I would suggest just to add `finally` clause.

**Comment by olsh** [comment]
Hi @vilchik-elena,

Thank you for the response.

The error was made not intentionally, moreover, the code compiles to working javascript.
From my point of view, the analyzer should fail with the parsing error and show what's wrong with the file, something like TSC does:
`error TS1005: 'finally' expected.`

**Comment by vilchik-elena** [comment]
I've created #578
